arch/src/filter_context.rs

```diff
@@ -282,7 +282,10 @@ impl RootContext for FilterContext {
         // No StreamContext can be created until the Embedding Store is fully initialized.
         let embedding_store = match self.mode {
             GatewayMode::Llm => None,
-            GatewayMode::Prompt => Some(Rc::clone(self.embeddings_store.as_ref().unwrap())),
```

Comment:
this unwrap can panic if a stream is being created before the embeddings store has finished initializing